galite-core/src/main/kotlin/org/kopi/galite/form/VForm.kt

```
@@ -736,6 +722,66 @@ abstract class VForm : VWindow, VConstants {
     return (getDisplay() as UForm).printForm()
   }
 
+  override fun executeVoidTrigger(VKT_Type: Int) {
+   return when (VKT_Type) {
+     3 -> {
+       postformAction()
+       super.executeVoidTrigger(VKT_Type)
+     }
+     4 -> {
+       initAction()
+       super.executeVoidTrigger(VKT_Type)
+     }
+     5 -> {
+       preformAction()
+       super.executeVoidTrigger(VKT_Type)
+     }
+    else -> {
+    }
+   }
+  }
+  fun executeObjectTrigger(VKT_Type: Int): Any {
+   return when (VKT_Type) {
+      3 -> {
+        postformAction()
+      }
+      4 -> {
+        initAction()
+      }
+      5 -> {
+        preformAction()
+      }
+      else -> {
+      }
+    }
+  }
+
+  fun executeBooleanTrigger(VKT_Type: Int): Boolean {
+    return when (VKT_Type) {
+      1 -> {
+        quitAction()
+      }
+      2 -> {
+        resetAction()
+      }
+      else -> {
+        true
```

throw `IllegalArgumentException` instead of returning true